presto-main/src/main/resources/webapp/src/components/QueryList.jsx

```diff
@@ -42,6 +42,34 @@ function getHumanReadableStateFromInfo(query) {
     );
 }
 
+
+function ResourceGroupLinks({groupId, length=35}) {
+    if (groupId && groupId.length > 0) {
+        var previousLen = 0;
+        var remaining = length;
+        const links = groupId.map((grp, idx) => {
+            remaining -= previousLen;
+            if (remaining <= 0) {
+                return null;
+            }
+            previousLen = grp.length + (idx === 0 ? 0 : 1);
+            return (
+                <a key={idx} className={idx === 0 ? '' : 'grouplink'} href={'./res_groups.html?group=' + encodeURIComponent(groupId.slice(0, idx + 1).join('.'))}>
```

Comment:
By breaking each react prop in a new line it will improve readability:
```
            <a 
                className={idx === 0 ? '' : 'grouplink'} 
                href={'./res_groups.html?group=' + encodeURIComponent(groupId.slice(0, idx + 1).join('.'))}
                key={idx} 
            >
                {truncateString(grp, remaining)}
            </a>
```